iree/turbine/kernel/wave/compile.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, List, Dict
+from .._support.indexing import IndexingContext, IndexExpr
+from ..compiler import kernel_codegen, host_codegen
+from .compile_options import WaveCompileOptions
+
+from .cache import (
+    is_cache_enabled,
+    get_cache_manager,
+    WAVE_RUNTIME_DIR,
+)
+from .utils.compile_utils import compile_to_vmfb
+from .utils.run_utils import invoke_vmfb, _write_file
+from iree.turbine.kernel._support.context import push, pop
+
+
+class WaveKernel:
+    """
+    Represents a wave kernel that can be invoked by the user.
+    """
+
+    def __init__(self, options: WaveCompileOptions, executable: Any, asm: str):
+        self.options = options
+        self.cached_kernel_hash = self.options.kernel_hash
+        self.subs = options.subs
+        self.executable = executable
+        self.asm = asm
+        # Load kernel from cache if available.
+        self.load_from_cache()
+
+    def load_from_cache(self):
+        cache_enabled = is_cache_enabled()
+        if cache_enabled:
+            cache_manager = get_cache_manager()
+            possible_cached_kernel = cache_manager.load_kernel(self.options.kernel_hash)
+            if possible_cached_kernel:
+                self.executable = possible_cached_kernel.vmfb
+
+    def __call__(self, *args, **kwargs):
+        return self.invoke(*args, **kwargs)
+
+    def invoke(self, *args, **kwargs):
+        """
+        Invokes the wave kernel with the given arguments.
+        Returns the assembly code of the compiled kernel.
+        """
+
+        # Partition arguments into kernel inputs and outputs.
+        kernel_inputs = []
+        kernel_outputs = []
+        for arg, b in zip(args, self.options.kernel_sig.kernel_buffer_bindings):
+            usage = b.kernel_buffer_type.usage
+            if usage == kernel_codegen.KernelBufferUsage.INPUT:
+                kernel_inputs.append(arg)
+
+            if usage == kernel_codegen.KernelBufferUsage.OUTPUT:
+                kernel_outputs.append(arg)
+
+        invoke_vmfb(self.executable, self.options, kernel_inputs, kernel_outputs)
```

Comment:
btw seems like the old kernel required us to feed in dynamic values/symbols, do we not need it any more here?
```py
    kernel_dynamic_dims = []
    if dynamic_symbols:
        kernel_dynamic_dims = dynamic_symbols_map.values()
```

Reply:
It is now contained inside the options so we don't need to explicitly pass it.

Reply:
nice!